server/schedulers/hot_region.go

```diff
@@ -755,16 +773,24 @@ func (bs *balanceSolver) betterThan(old *solution) bool {
 	if bs.cur.srcPeerStat != old.srcPeerStat {
 		// compare region
 
+		storeWeight := calcStoreWeight(
+			bs.cluster.GetStore(bs.cur.srcStoreID).GetRegionWeight(),
+			bs.cluster.GetStore(bs.cur.srcStoreID).GetLeaderWeight())
+
+		oldStoreWeight := calcStoreWeight(
+			bs.cluster.GetStore(old.srcStoreID).GetRegionWeight(),
+			bs.cluster.GetStore(old.srcStoreID).GetLeaderWeight())
+
 		if bs.rwTy == write && bs.opTy == transferLeader {
 			switch {
-			case bs.cur.srcPeerStat.GetKeyRate() > old.srcPeerStat.GetKeyRate():
+			case bs.cur.srcPeerStat.GetKeyRate()/storeWeight > old.srcPeerStat.GetKeyRate()/oldStoreWeight:
 				return true
-			case bs.cur.srcPeerStat.GetKeyRate() < old.srcPeerStat.GetKeyRate():
+			case bs.cur.srcPeerStat.GetKeyRate()/storeWeight < old.srcPeerStat.GetKeyRate()/oldStoreWeight:
 				return false
 			}
 		} else {
-			byteRkCmp := rankCmp(bs.cur.srcPeerStat.GetByteRate(), old.srcPeerStat.GetByteRate(), stepRank(0, 100))
-			keyRkCmp := rankCmp(bs.cur.srcPeerStat.GetKeyRate(), old.srcPeerStat.GetKeyRate(), stepRank(0, 10))
+			byteRkCmp := rankCmp(bs.cur.srcPeerStat.GetByteRate()/storeWeight, old.srcPeerStat.GetByteRate()/oldStoreWeight, stepRank(0, 100))
```

Comment:
maybe do it in `stepRank`? `compareSrcStore` and `compareDstStore` also need it.

Reply:
I didn't implement this in the stepRank method, because stepRank might become something like the following:
```
func stepRank(rk0 float64, step float64) func(float64) int64 {
   return func(rate storeWeight float64) int64 {
      return int64((rate/storeWeight - rk0) / step)
   }
}
```
This makes all the methods related to stepRank need to be changed,such as stLdByteRate,stLdKeyRate,sliceLoadCmp,sliceLPCmp. I personally think that such changes will increase the complexity of these functions. But I still tried to optimize this logic and added a getRateWithStoreWeight function.

what do you think?^_^